### ISSUE TITLE
If the default fake certificate is created on startup then the actual "default certificate" specified on the cli is never used / updated

### ISSUE THREAD

**Issue by primeroz** [jcmoraisjr/haproxy-ingress#618]
Note, this worked as expected in version 0.7.x , i am now upgrading to version 0.9 of the controller.

I use cert-manager to generate the default wildcard certificate for my domain and i specify it on the CLI , on a brand new cluster ( on my CI actually ) when haproxy starts the secret still does not exist since cert-manager did not finish creating it so i can see 

```
W0714 08:29:12.998359       8 controller.go:182] using auto generated fake certificate due to an error reading default TLS certificate: secret ingress-controller/star-letsencrypt-cert was not found
...
I0714 08:29:28.532994       8 backend_ssl.go:73] adding secret ingress-controller/star-letsencrypt-cert to the local store
I0714 08:31:38.217845       8 backend_ssl.go:65] updating secret ingress-controller/star-letsencrypt-cert in the local store
```

So the ingress start with the `fake default cert` and then , when the secret is created / updated by cert-manager the haproxy does pull it and add it to its local store. 

```
/ # ls -la /ingress-controller/ssl/
total 28
drwxr-xr-x    2 root     root          4096 Jul 14 08:31 .
drwxr-xr-x    1 haproxy  haproxy       4096 Jul 14 08:29 ..
-rw-------    1 root     root          2933 Jul 14 08:29 default-fake-certificate.pem
-rw-------    1 root     root          5274 Jul 14 08:31 ingress-controller_star-letsencrypt-cert.pem
```

Unfortunately though i can see that even after this happen the crt map is not updated to include the cert-manager generated secret

```
/ # cat /etc/haproxy/maps/_front001_bind_crt.list
# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
# #
# #   HAProxy Ingress Controller
# #   --------------------------
# #   This file is automatically updated, do not edit
# #
#
/ingress-controller/ssl/default-fake-certificate.pem
```

If i kill / restart the pod it has the correct secret in it

```
/ # cat /etc/haproxy/maps/_front001_bind_crt.list
# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
# #
# #   HAProxy Ingress Controller
# #   --------------------------
# #   This file is automatically updated, do not edit
# #
#
/ingress-controller/ssl/ingress-controller_star-letsencrypt-cert.pem
```

**Comment by primeroz** [comment]
I can see this behavior was added in 0.8 https://github.com/jcmoraisjr/haproxy-ingress/blob/v0.8.4/pkg/controller/controller.go#L131 

I kind of relied on the old behavior where the controller restarted until the default secret in kubernetes was actually available ... now i need to restart the whole deployment once the secret is created which is not very gitops friendly .

The only alternative i see right now is to specify the `secretName` on every ingress where i relied on the default one so that i will get smething like 

```
/etc/haproxy # cat maps/_front001_bind_crt.list 
# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # ## # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
# #
# #   HAProxy Ingress Controller# #   --------------------------
# #   This file is automatically updated, do not edit
# #
#
/ingress-controller/ssl/default-fake-certificate.pem
/ingress-controller/ssl/ingress-controller_star-letsencrypt-cert.pem dex.svc.ci1.gke-ci.mintelcloud.io
```  

Would you consider adding an option to restore the old behavior or have the controller update the default certificate once the secret is available ?

**Comment by jcmoraisjr** [comment]
Hi, yes this need to be fixed. Thanks for the report.

**Comment by primeroz** [comment]
@jcmoraisjr thank you! 

i can see this was added to v0.8 milestone.  Will it be part of 0.9 as well i assume ? :) i thought that was the current stable 

thanks again, for now i just added a secretName to every ingress and it looks to work ok

**Comment by jcmoraisjr** [comment]
This will be merged to master/v0.11 and cherry-picked up to v0.8. New release coming soon.

**Comment by jcmoraisjr** [comment]
Fixed in v0.8.5, v0.9.1 and v0.10-beta.3. Closing.
